src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -375,6 +380,33 @@ public int compareOffsets(final NakadiCursor first, final NakadiCursor second) t
         return KafkaCursor.fromNakadiCursor(first).compareTo(KafkaCursor.fromNakadiCursor(second));
     }
 
+    public long totalEventsInPartition(final Timeline timeline, final String partitionString) {
+        final Timeline.StoragePosition positions = timeline.getLatestPosition();
+        if (positions == null) {
+            throw new InvalidCursorOperation(InvalidCursorOperation.Reason.PARTITION_NOT_FOUND);
+        }
+        final int partition = Integer.valueOf(partitionString);
+        final Timeline.KafkaStoragePosition kafkaPositions = (Timeline.KafkaStoragePosition) positions;
+        final List<Long> offsets = kafkaPositions.getOffsets();
+        if (offsets.size() - 1 < partition) {
+            throw new InvalidCursorOperation(InvalidCursorOperation.Reason.PARTITION_NOT_FOUND);
+        } else {
+            return offsets.get(partition);
+        }
+    }
+
+    public long numberOfEventsBeforeCursor(final NakadiCursor cursor) {
+        return kafkaOffset(cursor);
+    }
+
+    public String getOffsetForPosition(final long offset) {
+        return StringUtils.leftPad(String.valueOf(offset), VERSION_ZERO_MIN_OFFSET_LENGTH, '0');
+    }
+
+    private long kafkaOffset(final NakadiCursor cursor) {
```

Comment:
it actually duplicates functionality of [KafkaCursors.toKafkaOffset](https://github.com/zalando/nakadi/blob/3d551da20b2fa4b5ed99a77a8aad0e120c38e3e6/src/main/java/org/zalando/nakadi/repository/kafka/KafkaCursor.java#L69)